src/openvic-simulation/scripts/ConditionalWeight.cpp

```diff
@@ -26,23 +26,76 @@ static NodeCallback auto expect_modifier(
 }
 
 node_callback_t ConditionalWeight::expect_conditional_weight(base_key_t base_key) {
-	return expect_dictionary_keys(
-		// TODO - add days and years as options with a shared expected count of ONE_EXACTLY
-		base_key_to_string(base_key), ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(base)),
-		"days", ZERO_OR_ONE, success_callback,
-		"years", ZERO_OR_ONE, success_callback,
+	key_map_t key_map;
+	bool successfully_set_up_base_keys = true;
+
+	switch (base_key) {
+	case BASE: {
+		successfully_set_up_base_keys &= add_key_map_entry(
+			key_map,
+			"base", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(base))
+		);
+	} break;
+	case FACTOR: {
+		successfully_set_up_base_keys &= add_key_map_entry(
+			key_map,
+			"factor", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(base))
+		);
+	} break;
+	case TIME: {
+		const auto time_callback = [this](std::string_view key, Timespan (*to_timespan)(Timespan::day_t)) -> auto {
+			return [this, key, to_timespan](uint32_t value) -> bool {
+				if (base == fixed_point_t::_0()) {
+					base = fixed_point_t::parse((*to_timespan)(value).to_int());
+					return true;
+				} else {
+					Logger::error(
+						"ConditionalWeight cannot have multiple base values - trying to set base to ", value, " ", key,
+						" when it already has a value equivalent to ", base, " days!"
+					);
+					return false;
+				}
+			};
+		};
+
+		successfully_set_up_base_keys &= add_key_map_entries(
+			key_map,
+			"days", ZERO_OR_ONE, expect_uint<uint32_t>(time_callback("days", Timespan::from_days)),
+			"months", ZERO_OR_ONE, expect_uint<uint32_t>(time_callback("months", Timespan::from_months)),
+			"years", ZERO_OR_ONE, expect_uint<uint32_t>(time_callback("years", Timespan::from_years))
+		);
+	} break;
+	default: {
+		successfully_set_up_base_keys = false;
+	}
```

Comment:
This format of switch is painfull for me to look at.
At line 70 & 71 we see a } but they are on very different levels and yet share the indentation.
At line 38, 44 & 67 the break is hidden after a } and isn't even inside the {}.
The default case doesn't even have a break.

This might just be cpp culture shock though.

Reply:
- There's no need for a break on the last case/default of a switch statement, all it does it prevent the code from continuing on to the next case (without a break the code will keep executing after hitting a valid case, regardless of other case keywords in its path)
- The code style of this repo has case keywords at the same indentation level as the swtich keyword, I've seen this a lot in C++ code. Curly brackets are often used to group each case's code more clearly, but I agree having them at the same indentation level as the outer brackets makes the code harder to read. I'll move the opening brackets onto a newline after each case and indent each case's code block one level beyond the switch and case keywords.